astropy/units/utils.py

```diff
@@ -229,32 +231,56 @@ def maybe_simple_fraction(p, max_denominator=100):
 
 
 def validate_power(p):
-    """Convert a power to a floating point value, an integer, or a Fraction.
+    """Check that a power can be converted to a floating point value.
+
+    Parameters
+    ----------
+    p : numerical
+        Power to be converted
+
+    Raises
+    ------
+    ValueError
+        If the power is an array in which not all elements are equal.
+    """
+    if p.__class__ is int or p.__class__ is Fraction:
+        return p
+    try:
+        float(p)
+    except Exception:
+        p = np.asanyarray(p)
+        if ((first := p.flat[0]) == p).all():
+            # All the same, now check it is OK.
+            float(first)
+            return first
+        else:
+            raise ValueError(
+                "Quantities and Units may only be raised to a scalar power"
+            ) from None
+    else:
+        return p
+
+
+def sanitize_power(p):
+    """Convert the power to a float, an integer, or a Fraction.
 
     If a fractional power can be represented exactly as a floating point
     number, convert it to a float, to make the math much faster; otherwise,
     retain it as a `fractions.Fraction` object to avoid losing precision.
     Conversely, if the value is indistinguishable from a rational number with a
     low-numbered denominator, convert to a Fraction object.
+    If a power can be represented as an integer, use that.
 
     Parameters
     ----------
     p : float, int, Rational, Fraction
-        Power to be converted
+        Power to be converted.
     """
+    if p.__class__ is int:
```

Comment:
any reason why not ?
```suggestion
    if p.__class__ is int or p.__class__ is Fraction:
```

Reply:
The purpose of this routine is also to simplify fractions to integer if the denominator equals 1 or o float if they can be represented exactly as that.

To be honest, I'm no longer all that sure that this great care of keeping fractions is all that helpful, but I didn't want to change things too much for a bug fix...

Reply:
fair enough !